prov/mlx/src/mlx_init.c

```diff
@@ -119,7 +118,13 @@ struct fi_fabric_attr mlx_fabric_attrs = {
 struct fi_ep_attr mlx_ep_attrs = {
 	.type = FI_EP_RDM,
 	.protocol = FI_PROTO_MLX,
-	.protocol_version = UCP_API_RELEASE,
+#if defined(UCP_API_RELEASE) && (UCP_API_RELEASE <= 2947)
+#warning "HPCX 1.9.7 have an issue with UCP_API_VERSION macro"
+	.protocol_version = (((UCP_API_MAJOR) << UCP_VERSION_MAJOR_SHIFT)|
+        ((UCP_API_MINOR) << UCP_VERSION_MINOR_SHIFT)),
+#else
+	.protocol_version = (UCP_API_VERSION),
```

Comment:
Isn't this using UCP_API_VERSION when it is NOT defined?

Reply:
Seems, [UCP_API_VERSION](https://github.com/openucx/ucx/blob/32869e6f8816bc0d8d34dc2eff77a1a287e7860d/src/ucp/api/ucp_version.h.in#L22) macro is always defined. But `UCP_API_RELEASE` is defined in `ucp/api/ucp_version.h` for release version of UCX.